File: backend/src/main/java/tom/projects/monkeylog/model/measurement/Measurement.java
package tom.projects.monkeylog.model.measurement;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;
import org.hibernate.annotations.UuidGenerator;
import tom.projects.monkeylog.model.MetricFormat;
import tom.projects.monkeylog.model.user.UserOwned;

import java.util.List;
import java.util.UUID;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Measurement implements UserOwned {
    @Id
    @GeneratedValue
    @UuidGenerator
    private UUID id;
    private String name;
    private UUID userId;
    @Enumerated(EnumType.STRING)
    private MetricFormat metric;
    @OneToMany(mappedBy = "measurement")
    @OnDelete(action = OnDeleteAction.CASCADE)
    private List<MeasurementPoint> measurementPoints;
}

File: backendkt/src/main/kotlin/tom/com/monkeylog/model/measurement/Measurement.kt
package tom.com.monkeylog.model.measurement

import jakarta.persistence.*
import org.hibernate.annotations.OnDelete
import org.hibernate.annotations.OnDeleteAction
import org.hibernate.annotations.UuidGenerator
import tom.com.monkeylog.model.MetricFormat
import tom.com.monkeylog.model.user.UserOwned
import java.util.*

@Entity
class Measurement(
    @Id
    @GeneratedValue
    @UuidGenerator
    var id: UUID,
    var name: String,
    override var userId: UUID?,
    @Enumerated(EnumType.STRING)
    var metric: MetricFormat,

    @OneToMany(mappedBy = "measurement")
    @OnDelete(action = OnDeleteAction.CASCADE)
    var measurementPoints: List<MeasurementPoint> = emptyList()
) : UserOwned